MapboxDirections/MBRouteOptions.swift

```diff
@@ -118,6 +124,13 @@ open class RouteOptions: DirectionsOptions {
      */
     @objc open var roadClassesToAvoid: RoadClasses = []
     
+    /**
+     Indicate how a route considers from which side of the road to approach a waypoint.
+     
+     By default, `ApproachType.unrestricted` is used meaning the route can approach a waypoint from either side of the road. If set to `ApproachType.curb`, the route will be returned so that on arrival, the waypoint will be found on the side that corresponds with the `DrivingSide` of the region in which the returned route is located.
```

Comment:
The fact that we end up documenting both of the enumeration’s cases here points strongly in the direction of replacing the enumeration with a simple Boolean-typed property. Additionally, this comment should explain _why_ the developer might want to set the option, not just what it does, and how this option differs from `allowsUTurnAtWaypoint`:

https://github.com/mapbox/MapboxDirections.swift/blob/038aa947fd137b9bd5c095eeb92cbaa0638fffab/MapboxDirections/MBRouteOptions.swift#L90